pkg/git/libgit2/managed/flag.go

```diff
@@ -21,14 +21,26 @@ import (
 	"strings"
 )
 
+var managedTransport bool
+
 // Enabled defines whether the use of Managed Transport should be enabled.
 // This is only affects git operations that uses libgit2 implementation.
 //
 // True is returned when the environment variable `EXPERIMENTAL_GIT_TRANSPORT`
 // is detected with the value of `true` or `1`.
 func Enabled() bool {
+	if managedTransport {
+		return true
+	}
+
 	if v, ok := os.LookupEnv("EXPERIMENTAL_GIT_TRANSPORT"); ok {
 		return strings.ToLower(v) == "true" || v == "1"
 	}
 	return false
 }
+
+// SetEnabled allows for the managed transport state to be changed on demand.
+// This is mainly used to enable testing.
```

Comment:
If this is just for testing, I wonder if it'd be better to just write a helper function that sets and unsets the env var instead of introducing conditionals in the main code that'll only be used for testing.

Reply:
That's a fair point, I removed the `SetEnabled` and just set the environment variable for the time being.

I haven't unset the environment variable as this has an ever lasting side effect. Once enabled, unmanaged transport is overwritten by the managed transport, so unsetting the variable may lead to an in-between state.

I am working towards having managed transport enabled by default soon (maybe next minor), so this will be flipped around.

Reply:
Thanks. Would be good to leave a comment about why it's not being unset at present. Usually we unset all the env vars set in tests.